src/utils/prepareAndValidate.ts

```diff
@@ -0,0 +1,17 @@
+import ResolutionError, {ResolutionErrorCode} from "../errors/resolutionError";
+
+/**
+  * Checks domain name for special symbols and returns address in lowercase without spaces
+  * @throws Will throw an error if domain address contains special symbols
+  * @param domain - a domain address
+  */
+const reg = RegExp('^[.a-z\\d-]+$');
```

Comment:
Are you sure that the regex is correct?
It doesn't seem to match domains with numbers. And it accepts the character `\` in a domain name.
https://regex101.com/r/dM7bP9/1

Maybe it should be `^[.a-z\d-]+$` without the extra `\`?